kernel/src/fs/kernfs/inode.rs

```diff
@@ -0,0 +1,447 @@
+        let mode = mode.unwrap_or(arc_parent.metadata().mode);
+        let metadata = Metadata::new_dir(ino, mode, BLOCK_SIZE);
+
+        let new_inode = Arc::new_cyclic(|weak_self| Self {
+            metadata: RwMutex::new(metadata),
+            elem: RwMutex::new(PseudoElement::new_dir()),
+            pseudo_extension,
+            fs: Arc::downgrade(&arc_parent.pseudo_fs()),
+            parent: Some(parent),
+            this: weak_self.clone(),
+        });
+        arc_parent.insert(name.to_string(), new_inode.clone())?;
+        Ok(new_inode)
+    }
+
+    pub fn new_symlink(
+        name: &str,
+        target: &str,
+        parent: Weak<PseudoINode>,
+        pseudo_extension: Option<Arc<dyn PseudoExt>>,
+    ) -> Result<Arc<Self>> {
+        let arc_parent = parent.upgrade().unwrap();
+        let ino = arc_parent.generate_ino();
+        let mode = InodeMode::from_bits_truncate(0o777);
+        let metadata = Metadata::new_symlink(ino, mode, BLOCK_SIZE);
+
+        let new_inode = Arc::new_cyclic(|weak_self| Self {
+            metadata: RwMutex::new(metadata),
+            elem: RwMutex::new(PseudoElement::new_symlink(target)),
+            pseudo_extension,
+            fs: Arc::downgrade(&arc_parent.pseudo_fs()),
+            parent: Some(parent),
+            this: weak_self.clone(),
+        });
+        arc_parent.insert(name.to_string(), new_inode.clone())?;
+        Ok(new_inode)
+    }
+
+    /// Get the current node.
+    pub fn this(&self) -> Arc<PseudoINode> {
+        self.this.upgrade().unwrap()
+    }
+
+    /// Get the current node as a weak reference.
+    pub fn this_weak(&self) -> Weak<PseudoINode> {
+        self.this.clone()
+    }
+
+    /// Get the parent of the current node.
+    fn parent(&self) -> Option<Arc<PseudoINode>> {
+        self.parent.as_ref().and_then(|p| p.upgrade())
+    }
+
+    /// Get the pseudo filesystem of the current node.
+    fn pseudo_fs(&self) -> Arc<dyn PseudoFileSystem> {
+        self.fs.upgrade().unwrap()
+    }
+
+    /// Generate a new inode number for the current node.
+    fn generate_ino(&self) -> u64 {
+        self.pseudo_fs().alloc_id()
+    }
+
+    /// Set the data of the current node.
+    pub fn set_data(&self, data: Box<dyn DataProvider>) -> Result<()> {
+        self.elem.write().set_data(data)
+    }
+
+    /// Set the pseudo_extension of the current node.
+    /// The pseudo_extension provides different abstractions for different pseudo-file system requirements.
+    pub fn set_pseudo_extension(&mut self, pseudo_extension: Arc<dyn PseudoExt>) {
+        self.pseudo_extension = Some(pseudo_extension);
+    }
+
+    /// Remove a child from the current node.
+    /// Return Ok(Arc<dyn Inode>) if the child is removed successfully.
+    /// Current node should be a directory.
+    /// Trigger the pseudo_extension's on_remove method if the pseudo_extension exists.
+    fn remove(&self, name: &str) -> Result<Arc<dyn Inode>> {
```

Comment:
Use consistent API doc-level comment phrase - `/// Removes a child...` The "this method" at the beginning is omitted. And use \`\` to wrap the code such as `on_remove`.